webapp/src/components/propertyValueElement.tsx

```diff
@@ -38,11 +39,15 @@ const PropertyValueElement = (props:Props): JSX.Element => {
             return !isNaN(parseInt(val, 10))
         case 'email': {
             const emailRegexp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
-            return emailRegexp.test(val.toLowerCase())
+            return emailRegexp.test(lowerCaseVal)
         }
         case 'url': {
             const urlRegexp = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w\-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)/
-            return urlRegexp.test(val.toLowerCase())
+            return urlRegexp.test(lowerCaseVal)
+        }
+        case 'phone': {
+            const phoneRegexp = /[+]?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6}/
```

Comment:
@chenilim how do you feel about this? I don't know how international is the way of writing phone numbers, looks more or less accurate, at least for Spain, and probably for US, but I dont't think is going to work always. For example "+34 666666666" doesn't work, and probably is a very common way of writting it (That should be +034 to make it work). Also, in Spain if you write a phone number with dots in it, probably you would use something like "666.666.666" or even "666.66.66.66" but never something like "666.666666". And that is only for Spain, I guess each country have its own slightly different ways of writing it.

Reply:
There are other kind of peculiarities to take into consideration, for example "extensions".

Reply:
Sorry for shooting in from the side here but I, too, made the mistake of trying to validate phone numbers with a regex once. Unfortunately, phone numbers are poorly standardized, especially when you factor in different formatting options for the same number. If you target a global audience and if the input has to be validated, I can recommend google/libphonenumber (or one of the forks that exist for various languages).

Reply:
Yes, I totally agree with @Johennes here, this is a hard problem and we should use something like the libphonenumber or just not validate it at all. The problem with libphonenumber is really heavyweight, around 420K because requires google clousere compiler, and there is tinier fork but still 110K. I'm not 100% sure what is the best option.

Reply:
Yes, probably this is the most sensible option if we want to have phone number validations: https://github.com/catamphetamine/libphonenumber-js

Reply:
Yeah, agreed @jespino , Thanks for the links!
We can either use libphonenumber-js or skip validation for phone number as you were suggesting. Let me know what you think. And I'll make the corresponding changes.

Reply:
@chenilim any strong opinion here? I'm more inclined to not-validate it for now, but if we want validation, looks like the more sane approach is the libphonenumber-js option.

Reply:
Sorry @jespino, I missed your earlier comment. Looked more into this. We should **remove validation** from the phone number. The main value of a phone number field is to (later) add a "call this number" button (makes most sense from mobile). However, the "number" may not actually be a number. E.g. it could be a Facetime contact or something similar.

So, no need to pull in another library for now. P4 (maybe in the far future), we could allow users to pick from a set of regex templates, and maybe show a warning rather than hard validation.

Reply:
Removed phone number validation @ https://github.com/mattermost/focalboard/pull/286/commits/d87b90e6b932e2cf7693871d26e9bad6a3efda42